packages/enr/src/multiaddrs_codec.ts

```diff
@@ -20,7 +20,16 @@ export function decodeMultiaddrs(bytes: Uint8Array): Multiaddr[] {
     const multiaddrBytes = bytes.slice(index, index + size);
     index += size;
 
-    multiaddrs.push(multiaddr(multiaddrBytes));
+    let ma = multiaddr(multiaddrBytes);
+
+    // if the multiaddr contains `p2p`, it means that it supports `p2p-circuit`
+    // this is avoided from the ENR to save space
+    // we should add it
+    if (ma.toString().includes("/p2p/")) {
```

Comment:
I think we should handle particular case because to me it seems possible to accidentally get `p2p/p2p-circuit` or something.

```ts
if (hasP2p && isPeerId(left) && isPeerId(right)) {
  ...
}
```

Correct me here if I am wrong.